docs/services/introspection.mdx

```diff
@@ -0,0 +1,357 @@
+
+    </TabItem>
+</Tabs>
+
+### Checking how an invocation was triggered
+To find out if an invocation was triggered via the ingress or by another service:
+
+<Tabs groupId="interface" queryString>
+    <TabItem value="cli" label="CLI">
+
+        ```shell
+        restate invocations describe <INVOCATION_ID>
+        ```
+
+        You can check this at the root of the tree in the invocation progress.
+
+    </TabItem>
+    <TabItem value="psql" label="psql">
+
+        ```sql
+        select invoked_by, invoked_by_service, invoked_by_id from sys_status where id = '<INVOCATION_ID>';
+        ```
+        The `invoked_by` field contains either `ingress` or `service`. If the invocation was triggered by another service, then the fields `invoked_by_service` and `invoked_by_id` will supply more information about the invoking service.
+
+    </TabItem>
+</Tabs>
+
+### Retrieving the trace ID of an invocation
+
+<Tabs groupId="interface" queryString>
+    <TabItem value="cli" label="CLI">
+
+        ```shell
+        restate invocations describe <INVOCATION_ID>
+        ```
+
+    </TabItem>
+    <TabItem value="psql" label="psql">
+
+        ```sql
+        select trace_id from sys_status where id = '<INVOCATION_ID>';
+        ```
+
+    </TabItem>
+</Tabs>
+
+  Afterwards, you can use this trace ID to [search for spans in Jaeger](/restate/tracing#searching-traces).
+
+### Listing inactive invocations
+
+<Tabs groupId="interface" queryString>
+    <TabItem value="cli" label="CLI">
+        To list the oldest invocations that are not making progress:
+
+        ```shell
+        restate invocations list --oldest-first --status pending,backing-off,suspended
+        ```
+
+    </TabItem>
+    <TabItem value="psql" label="psql">
+        To retrieve all invocations that have not seen any activity for more than 1 hour:
+
+        ```sql
+        select * from sys_status where to_timestamp(modified_at) <= now() - interval '1' hour;
+        ```
+
+    </TabItem>
+</Tabs>
+
+### Listing zombie invocations
+
+Zombie invocations are invocations that are pinned to a specific deployment but that deployment was forcefully removed. You can list them by executing:
+
+<Tabs groupId="interface" queryString>
+    <TabItem value="cli" label="CLI">
+
+        ```shell
+        restate inv ls --zombie
+        ```
+
+    </TabItem>
+    <TabItem value="psql" label="psql">
+        This is not possible via only `psql`.
+
+        If you know the ID of the deployment that got forcefully removed, you can check if there are zombie invocations pinned to it via:
+
+        ```sql
+        select * from sys_status where pinned_deployment_id = '<DEPLOYMENT_ID>';
+        ```
+
+    </TabItem>
+</Tabs>
+
+## Inspecting application state
+
+:::note
+Inspecting the application state is currently only possible via the CLI.
```

Comment:
Inverted? it's currently only possible through the psql interface, not the CLI, right?

Reply:
Good catch